src/entity.cpp

```diff
@@ -9226,3 +9261,82 @@ void Entity::setRangedProjectileAttack(Entity& marksman, Stat& myStats)
 		}
 	}
 }
+
+/* entity.cpp
+ * Attempts to set the Entity on fire. Entities that are not Burnable or are already on fire will return before any processing
+ * Entities that do not have Stats (such as furniture) will return after setting the fire time and chance to stop at max
+ * Entities with Stats will have their fire time (char_fire) and chance to stop being on fire (chanceToPutOutFire) reduced by their CON
+ * Calculations for reductions is outlined in this function
+ */
+void Entity::SetEntityOnFire()
+{
+	// Check if the Entity can be set on fire
+	if ( this->flags[BURNABLE] )
+	{
+		// Check if the Entity is already on fire
+		if ( !(this->flags[BURNING]) )
+		{
+			this->flags[BURNING] = true;
+			serverUpdateEntityFlag(this, BURNING);
+
+			/* Set the time the Entity will be on fire, based off their CON
+			 * |\_ MAX_TICKS_ON_FIRE is reduced by every 2 points in CON
+			 * |
+			 * |\_ Fire has a minimum of 2 seconds (100 ticks), and a maximum of 20 seconds (1000 ticks)
+			 * |  \_ Constants are defined in entity.hpp: MIN_TICKS_ON_FIRE and MAX_TICKS_ON_FIRE
+			 * |
+			 *  \_ For every 5 points of CON, the chance to stop being on fire is reduced
+			 *    \_ The chance to stop being on fire has a minimum of 1 in 10, and a maximum of 1 in 5
+			 *      \_ Constants are defined in entity.hpp: MIN_CHANCE_STOP_FIRE and MAX_CHANCE_STOP_FIRE
+			 */
+
+			// Set the default time on fire
+			this->char_fire = MAX_TICKS_ON_FIRE;
+			// Set the default chance of putting out fire
+			this->chanceToPutOutFire = MAX_CHANCE_STOP_FIRE;
+
+			// If the Entity is not a Monster, it wont have Stats, end here
+			if ( this->getStats() == nullptr )
+			{
+				return; // The Entity was set on fire, it does not have Stats, so it is on fire for maximum duration
+			}
+
+			// Determine decrease in time on fire based on the Entity's CON
+			const Sint32 entityCON = this->getStats()->CON;
+
+			// If the Entity's CON is <= 1 then their time is just MAX_TICKS_ON_FIRE
+			if ( entityCON <= 1 )
+			{
+				return; // The Entity was set on fire, with maximum duration and chance
+			}
+
+			// If the Entity's CON is <= 4 then their chance is just MAX_CHANCE_STOP_FIRE
+			if ( entityCON <= 4 )
+			{
+				this->chanceToPutOutFire = MAX_CHANCE_STOP_FIRE;
+			}
+			else if ( entityCON >= 25 ) // If the Entity has 25 or greater CON, then the reduction is equal to or less than MIN_CHANCE_STOP_FIRE
```

Comment:
Shouldn't this line and line 9314 not be hardcoded? Rather, calculating the cutoffs via the actual value of those two constants.